autorest/adal/cli_token.go

```diff
@@ -0,0 +1,32 @@
+package adal
+
+import "strconv"
+
+type AzureCLIToken struct {
+	AccessToken      string `json:"accessToken"`
+	Authority        string `json:"_authority"`
+	ClientID         string `json:"_clientId"`
+	ExpiresIn        int     `json:"expiresIn"`
+	ExpiresOn        string `json:"expiresOn"`
+	IdentityProvider string `json:"identityProvider"`
+	IsMRRT           bool    `json:"isMRRT"`
+	RefreshToken     string `json:"refreshToken"`
+	Resource         string `json:"resource"`
+	TokenType        string `json:"tokenType"`
+	UserId           string `json:"userId"`
+}
+
+func AzureCLIAccessTokensPath() string {
+	return "~/.azure/accessTokens.json"
```

Comment:
Awesome, thanks Tom. This looks like it'll do the trick!

@jhendrixMSFT, @mcardosos: I verified that this library is using the MIT license (which is a very permissive license) and registered our usage of it internally.